vision_agent/agent/vision_agent.py

```diff
@@ -553,14 +500,6 @@ def chat_with_artifacts(
                         code_interpreter,
                     )
                     obs_chat_elt: Message = {"role": "observation", "content": obs}
-                    media_obs = check_and_load_image(code_action)
-                    if media_obs and result.success:
-                        # media paths will be under the local_save_path when we download
-                        # them after each turn
-                        obs_chat_elt["media"] = [
-                            artifacts.local_save_path.parent / media_ob
-                            for media_ob in media_obs
-                        ]
```

Comment:
Here if the agent calls `view_media_artifact`, we then add that to the media key of the current chat so the agent can actually see the image. I don't see this case being handled anywhere with these lines deleted